tools/docvalidator/main_test.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// MockFileSystem is a mock implementation of the FileSystem interface for testing purposes.
+type MockFileSystem struct {
+	Files map[string]string
+}
+
+func (m *MockFileSystem) Glob(pattern string) ([]string, error) {
+	var matched []string
+	for file := range m.Files {
+		matched = append(matched, file)
+		if strings.Contains(file, "error_glob") {
+			return nil, errors.New("error matching files")
+		}
+	}
+	return matched, nil
+}
+
+func (m *MockFileSystem) ReadFile(filename string) ([]byte, error) {
+	content, ok := m.Files[filename]
+	if !ok {
+		return nil, errors.New("file not found")
+	}
+	if strings.Contains(filename, "error_read") {
+		return nil, errors.New("error reading file")
+	}
+
+	return []byte(content), nil
+}
+
+// TestFilterFiles tests that filterFiles correctly removes specified files from the list.
+func TestFilterFiles(t *testing.T) {
+	files := []string{
+		"docs/registries/example1.md",
+		"docs/registries/list-of-all-registries.md",
+		"docs/registries/example2.md",
+		"docs/registries/backward.md",
+	}
+	remove := []string{
+		"list-of-all-registries.md",
+		"backward.md",
+	}
+	expected := []string{
+		"docs/registries/example1.md",
+		"docs/registries/example2.md",
+	}
+	result := filterFiles(files, remove)
+	assert.Equal(t, expected, result)
+}
+
+func TestRun_ErrorSettingLogLevel(t *testing.T) {
+	logLevel = "invalid"
+	defer func() { logLevel = "info" }()
+
+	fs := &MockFileSystem{
+		Files: map[string]string{},
+	}
+
+	err := run(fs)
+
+	assert.ErrorContains(t, err, "invalid log level: invalid")
+}
+
+// TestRun_NoFiles tests run function when no markdown files are found.
+func TestRun_NoFiles(t *testing.T) {
+	fs := &MockFileSystem{
+		Files: map[string]string{},
+	}
+
+	err := run(fs)
+	require.Error(t, err)
+	assert.Equal(t, "no Markdown files found in docs directory", err.Error())
```

Comment:
This is EqualError then

But usually, either you use ErrorContains, or you return a sentinel error and use ErrorIs here to detect it